docs/source/guides/testing-react-apollo.md

```diff
@@ -0,0 +1,334 @@
+    {(mutate, { loading, error, data }) => {
+      if (loading) return <p>Loading...</p>;
+      if (error) return <p>Error!</p>;
+      if (data) return <p>Deleted!</p>;
+
+      return (
+        <button onClick={() => mutate({ variables: { name: 'Buck' } })}>
+          Click me to Delete Buck!
+        </button>
+      );
+    }}
+  </Mutation>
+);
+```
+
+Testing an initial render for this component looks identical to testing our `Query` component.
+
+```js
+import DeleteButton, { DELETE_DOG_MUTATION } from './delete-dog';
+
+it('should render without error', () => {
+  renderer.create(
+    <MockedProvider mocks={[]}>
+      <DeleteButton />
+    </MockedProvider>,
+  );
+});
+```
+
+Calling the mutation is where things get interesting:
+
+```js
+it('should render loading state initially', () => {
+  const deleteDog = { name: 'Buck', breed: 'Poodle', id: 1 };
+  const mocks = [
+    {
+      request: {
+        query: DELETE_DOG_MUTATION,
+        variables: { name: 'Buck' },
+      },
+      result: { data: { deleteDog } },
+    },
+  ];
+
+  const component = renderer.create(
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <DeleteButton />
+    </MockedProvider>,
+  );
+
+  // find the button and simulate a click
+  const button = component.root.findByType('button');
+  button.props.onClick(); // fires the mutation
+
+  const tree = component.toJSON();
+  expect(tree.children).toContain('Loading...');
+});
+```
+
+This example looks very similar to the `Query` component. The difference comes after the render. Since this component relies on a button to be clicked to fire a mutation, we first use the renderer's API to find that button.
+
+After we find the button, we can manually call the `onClick` prop to that button, simulating a user clicking the button. This click fires off the mutation, and then the rest is tested identically to the `Query` component.
+
+For example, to test for a successful mutation, you'd just wait for the promise to resolve from `MockedProvider` and check for any confirmation message in your UI, just like the `Query` component:
+
+```js
+it('should delete and give visual feedback', async () => {
+  const deleteDog = { name: 'Buck', breed: 'Poodle', id: 1 };
+  const mocks = [
+    {
+      request: {
+        query: DELETE_DOG_MUTATION,
+        variables: { name: 'Buck' },
+      },
+      result: { data: { deleteDog } },
+    },
+  ];
+
+  const component = renderer.create(
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <DeleteButton />
+    </MockedProvider>,
+  );
+
+  // find the button and simulate a click
+  const button = component.root.findByType('button');
+  button.props.onClick(); // fires the mutation
+
+  await wait(0);
+
+  const tree = component.toJSON();
+  expect(tree.children).toContain('Deleted!');
+});
+```
+
+Testing UI components isn't a simple issue, but hopefully with these tools, you can feel more confident when testing components dependent on data.
```

Comment:
Ideally, yeah! I wrote a test for doing that in the codesandbox example I linked to at the bottom.

I commented it out because for some reason when testing, it would throw the error on the preview window (even though the component is never mounted).

I didn't have the time to figure out what was going on there, unfortunately